test/manipulation.test.js

```diff
@@ -1576,6 +1576,34 @@ describe('manipulation', function() {
         done();
       });
     });
+
+    it('correctly coerces the PK value', async () => {
+      const created = await Post.create({
+        title: 'a title',
+        content: 'a content',
+      });
+
+      // Emulate what happens when model instance is received by REST API clients
+      const data = JSON.parse(JSON.stringify(created));
+
+      // Modify some of the data
+      data.title = 'Draft';
+      delete data.content;
+
+      // Call replaceById to modify the database record
+      await Post.replaceById(data.id, data);
+
+      // Verify what has been stored
+      const found = await Post.findById(data.id);
+      found.toObject().should.eql({
+        id: created.id,
+        title: 'Draft',
+        content: undefined,
+      });
+
+      // Verify that no warnings were triggered
+      Object.keys(Post._warned).should.be.empty();
```

Comment:
Without my fix in place, this checks passes when we run the tests in juggler, but fails against the MongoDB connector. I have verified that myself by running the MongoDB test suite against `npm link`-ed version of juggler.